docs/extensions.rst

```diff
@@ -133,3 +133,38 @@ TLS
 ---
 
 See :doc:`specs/tls`.
+
+HTTP Trailers
+---
+
+The Trailer response header allows the sender to include additional fields at the
+end of chunked messages in order to supply metadata that might be dynamically
+generated while the message body is sent, such as a message integrity check,
+digital signature, or post-processing status.
+
+ASGI servers that implement this extension will provide
+``http.response.trailers`` in the extensions part of the scope::
+
+    "scope": {
+        ...
+        "extensions": {
+            "http.response.trailers": {},
+        },
+    }
+
+An ASGI framework interested in sending trailing headers to the client, must include
+the ``Trailer`` header in the ``http.response.start``. That will allow the ASGI server
+to know that after the last ``http.response.body`` message (``more_body`` being ``False``),
+the ASGI framework will send a ``http.response.trailers`` message.
+
+Keys:
+
+* ``type`` (*Unicode string*): ``"http.response.trailers"``
+
+* ``headers`` (*Iterable[[byte string, byte string]]*): An iterable of
+  ``[name, value]`` two-item iterables, where ``name`` is the header name, and
+  ``value`` is the header value. Header names must be lowercased. Pseudo
+  headers (present in HTTP/2 and HTTP/3) must not be present.
+
+The ASGI server will only send the trailing headers in case the client has sent the
+``TE: trailers`` header in the request.
```

Comment:
Should the server error on receipt of this message without the `TE: trailers` header or just silently not send the trailing headers?

Reply:
I think it should be more on the "silently not send" side.

Reply:
> Should the server error on receipt of this message without the `TE: trailers` header or just silently not send the trailing headers?

In my implementation, the protocol server just send trailers when it receive that event. I think it's the responsibility of the asgi application to decide whether to send or not. So I don't have ```trailer``` field in ```http.response.start```. If you want to send, just send.

Reply:
I did it the same way in my separate PR. I agree the responsibility should be on the asgi app